client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -176,6 +180,10 @@ int getNumberOfUnassignedSegments() {
         return new HashMap<>(unassignedSegments);
     }
 
+    Map<Segment, Long> getEndSegments() {
+        return Collections.unmodifiableMap(new HashMap<>(endSegments));
```

Comment:
ImmutableMap.of()

Reply:
Updated code, since `endSegments` is an ImmutableMap we can return `endSegments` directly here. (ImmutableMap.copyOf(endSegments) also returns endSegments.